pkg/apis/servicecatalog/validation/serviceplan.go

```diff
@@ -65,6 +65,11 @@ func internalValidateServicePlan(serviceplan *sc.ServicePlan) field.ErrorList {
 
 func validateServicePlanSpec(spec *sc.ServicePlanSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	if "" == spec.ServiceBrokerName {
```

Comment:
If we let the user set the `ServiceBrokerName` manually, we should validate that the `ServiceClass` (plan's parent) belongs to the same broker.

Reply:
I don't think we need to make special accomodation for setting this field.  Ordinary users will not create this resource.  We _should_ have a referential integrity check while references are being resolved, but I do not think we should make referential integrity checks at creation time.  This is the exact same as setting the same field on the ServiceClass resource.